src/python/zigopt/experiment/progress.py

```diff
@@ -92,10 +92,10 @@ def _observation_progress_for_experiments(self, experiments, should_fetch_best,
         func.max(Observation.id),
         func.min(Observation.id),
         func.count(Observation.id),
-        func.sum(Observation.data.task.cost.as_numeric()),  # type: ignore
+        func.sum(Observation.data.task.cost.as_primitive()),  # type: ignore
```

Comment:
why does this `task.cost` need `as_primitive()` but the one below (L144) not?

Reply:
We can only do the automatic cast when operating on a column. `Observation.data.task.cost == 1` is an equality operation, so the automatic casting (and default value) is performed immediately before the equality check. I tried to cast on JSONB access (essentially when accessing the `cost` attribute of `task`), but this causes issues in other places, specifically when calling update:
```python
{
  Observation.data.task.cost: 1,
}
```
In this case we try to unwind the JSON path of `Observation.data.task.cost` but `Observation.data.task.cost` actually looks like `cast(coalesce(Observation.data.task.cost, :default_cost), NUMERIC)`, so our code for unwinding the JSON path doesn't work anymore.